src/App.tsx

```diff
@@ -380,6 +540,73 @@ export default class App extends React.Component<{}, { expandFolderBrowser: bool
     await this.loginComponent.current?.refreshTokenAsync(true);
   }
 
+  getShouldShowSelect(): boolean {
+    return !this.entrySelected && !!this.repositoryBrowser?.current?.currentFolder;
```

Comment:
getter doesn't really get updated with changes, so I made those properties states, and set the states to those values when changes happen. this getShouldShowSelect is not a getter but is used to set the values. I should change the name to setShouldShowSelect and use setState inside

Reply:
I mean it looks like you just call this when you want the value.

Instead of
const shouldSelect = this.getshouldShowSelect();

you can jsut do this.shouldShowSelect (and the value would be the same function)

Reply:
shouldShowSelect doesn't get updated in the UI if it is a property of the class. so I made it a state